### PR TITLE
Test invalid parameters in imageflip

### PR DESCRIPTION
**Pull request by vitormattos**
Test invalid parameters in imageflip

By @PHPRio
In TestFest 2017

### CODE REVIEW

File: ext/gd/tests/imageflip_error.phpt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+var_dump(imageflip($im, 'invalid_mode'));
+var_dump(imageflip(fopen(__FILE__, 'r'), IMG_FLIP_BOTH));
+define('IMG_FLIP_INVALID', 9999);
+var_dump(imageflip($im, IMG_FLIP_INVALID));
+?>
+--EXPECTF--
+Warning: imageflip() expects parameter 1 to be resource, null given in %s on line %d
+NULL
+
+Warning: imageflip() expects parameter 2 to be integer, string given in %s on line %d
```
Comment (krakjoe) [inline ext/gd/tests/imageflip_error.phpt:35]: This is an unnecessary test of zpp.

**Review by krakjoe — CHANGES_REQUESTED** [review]
zpp

File: ext/gd/tests/imageflip_error.phpt (hunk 1 above)
Comment (raphaeldealmeida) [reply ext/gd/tests/imageflip_error.phpt:35]: @krakjoe Our proposition was to cover fully imageflip function (https://github.com/phpcommunity/phptestfest-php-src/blob/phptestfest-2017-master/ext/gd/gd.c#L4640)

Please help us to create better tests in this case.

File: ext/gd/tests/imageflip_error.phpt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+NULL
+
+Warning: imageflip(): supplied resource is not a valid Image resource in %s on line %d
+bool(false)
+
+Warning: imageflip(): Unknown flip mode in %s on line %d
```
Comment (carusogabriel) [inline ext/gd/tests/imageflip_error.phpt:41]: Please remove the other ZPP failure test and keep this one.

**Review by carusogabriel — CHANGES_REQUESTED** [review]

File: ext/gd/tests/imageflip_error.phpt (hunk 1 above)
Comment (cmb69) [reply ext/gd/tests/imageflip_error.phpt:35]: @raphaeldealmeida See <https://externals.io/message/103886>.

**Comment by petk** [comment]
Thank you for the pull request. Since we can't know the status of this pr anymore, I'll be merging the rest of the contributions in a single commit... More info on this soon.

**Pull request closed without merging**